### PR TITLE
userspace: Enable build of both A and B images.

### PR DESCRIPTION
**Pull request by osenft**
This change adds the option for Rust userspace apps to be built with
both and A and a B image with different address offsets.

It also enables this functionality for the otpilot app.

### CODE REVIEW

File: kernel/Build.mk [hunk 1]
```diff
@@ -12,12 +12,31 @@
...
+kernel/build$(IMAGE): sandbox_setup
+	cd kernel && \
+		CARGO_TARGET_DIR="../build/kernel/cargo$(IMAGE)" \
+		RUSTFLAGS="-C link-arg=-T./layout$(IMAGE).ld" \
+		$(BWRAP) cargo build --release
+
+.PHONY: kernel/build-signed$(IMAGE)
+kernel/build-signed$(IMAGE): kernel/build$(IMAGE)
+
+endef
```
Comment (dann21) [inline kernel/Build.mk]: Pernaps ` endef # KERNEL_IMAGE_TARGETS` would make this more readable?

File: userspace/Build.mk [hunk 2]
```diff
@@ -279,42 +271,68 @@ userspace/$(APP)/$(BOARD)/doc: sandbox_setup build/gitlongtag
...
-userspace/$(APP)/$(BOARD)/program: \
+endef
+
+# ------------------------------------------------------------------------------
+# Macro to define targets for a specific board-app-and-image combination.
+# Arguments:
+# - $(BOARD)
+# - $(APP)
+# - $(IMAGE) [optional suffix to build A/B images]
+define RUST_APP_BOARD_IMAGE_TARGETS
```
Comment (dann21) [inline userspace/Build.mk:282]: IMO there ought to be a comment attached to the closing `endef` for this `define` that makes it easy for the reader to visually discern the entirety of the macro body.

**Review by dann21 — APPROVED** [review]
LGTM -- a couple of comment nits, that's all.

File: kernel/Build.mk (hunk 1 above)
Comment (osenft) [reply kernel/Build.mk]: Good call. Done throughout.

File: userspace/Build.mk (hunk 2 above)
Comment (osenft) [reply userspace/Build.mk:282]: Done throughout.

**Review by dann21 — APPROVED** [review]
LGTM (modulo build failure)

**Review by jrvanwhy — APPROVED** [review]

**Pull request merged**